### PR TITLE
Add `"monitor": "any"|"toCurrent"|"toMouse"` setting to globalSummon

### PR DESCRIPTION
**Pull request by zadjii-msft**
## Summary of the Pull Request

This adds support for part of the `monitor` property for `globalSummon`. It also goes a little off-spec:

```json
"monitor": "any"|"toCurrent"|"toMouse"
```

* `monitor`: This controls the monitor that the window will be summoned from/to
  - `"any"`: Summon the MRU window, regardless of which monitor it's currently on.
  - `"toCurrent"`/omitted: (_default_): Summon the MRU window **TO** the monitor with the current **foreground** window.
  - [**NEW**] `"toMouse"`: Summon the MRU window **TO** the monitor where the **mouse** cursor is.

When I was playing with this, It felt like `toMouse` was always what I wanted, not `toCurrent`. We can always just comment that out if we think that's contentious - I'm aware I didn't originally spec that.

## References
* Original thread: #653
* Spec: #9274 
* megathread: #8888

## PR Checklist
* [x] Closes https://github.com/microsoft/terminal/projects/5#card-60325291
* [x] I work here
* [ ] Tests added/passed
* [ ] Requires documentation to be updated 😢 

## Detailed Description of the Pull Request / Additional comments

I made `toMouse` the default because it felt better. fite-me.jpg 

## Validation Steps Performed
my ever evolving blob:

```jsonc
    { "keys": "ctrl+`", "command": { "action": "quakeMode" } },
    { "keys": "ctrl+1", "command": { "action": "globalSummon" } },
    // { "keys": "ctrl+2", "command": { "action": "globalSummon", "desktop": "toCurrent" } },
    // { "keys": "ctrl+2", "command": { "action": "globalSummon", "toggleVisibility": false } },
    // { "keys": "ctrl+2", "command": { "action": "globalSummon", "dropdownDuration": 2000 } },
    { "keys": "ctrl+2", "command": { "action": "globalSummon", "monitor": "any" } },
    // { "keys": "ctrl+3", "command": { "action": "globalSummon", "desktop": "onCurrent" } },
    { "keys": "ctrl+3", "command": { "action": "globalSummon", "monitor": "toMouse" } },
    // { "keys": "ctrl+4", "command": { "action": "globalSummon", "desktop": "any" } },
    { "keys": "ctrl+4", "command": { "action": "globalSummon", "monitor": "toMouse", "dropdownDuration": 500 } },
    { "keys": "ctrl+5", "command": { "action": "globalSummon", "dropdownDuration": 500 } },
```

### CODE REVIEW

**Comment by miniksa** [comment]
Man I make one comment in bug bash and here it is! Nice!

File: src/cascadia/WindowsTerminal/AppHost.cpp [hunk 1]
```diff
@@ -695,6 +699,19 @@ void AppHost::_GlobalHotkeyPressed(const long hotkeyIndex)
             args.SummonBehavior().ToggleVisibility(summonArgs.ToggleVisibility());
             args.SummonBehavior().DropdownDuration(summonArgs.DropdownDuration());
 
+            switch (summonArgs.Monitor())
+            {
+            case Settings::Model::MonitorBehavior::Any:
```
Comment (miniksa) [inline src/cascadia/WindowsTerminal/AppHost.cpp:704]: Annoying to have to do this but I suspect it's just so you don't cross dependencies between remoting and settings model, yeah?

**Review by miniksa — APPROVED** [review]

File: src/cascadia/WindowsTerminal/AppHost.cpp (hunk 1 above)
Comment (zadjii-msft) [reply src/cascadia/WindowsTerminal/AppHost.cpp:704]: it absolutely 100% is. In the future, there will be `TSM::MonitorBehavior::OnCurrent`, which will actually translate to `Remoting...InPlace` as well, so it's not totally 1-1 (though these 3 totally are)

Comment (miniksa) [reply src/cascadia/WindowsTerminal/AppHost.cpp:704]: Good enough. Thanks.

File: src/cascadia/WindowsTerminal/AppHost.cpp [hunk 2]
```diff
@@ -589,9 +589,13 @@ bool AppHost::HasWindow()
...
                                    Remoting::CommandlineArgs args)
 {
+    const Remoting::SummonWindowBehavior summonArgs{};
+    summonArgs.MoveToCurrentDesktop(false);
+    summonArgs.DropdownDuration(0);
+    summonArgs.ToMonitor(Remoting::MonitorBehavior::InPlace);
     // Summon the window whenever we dispatch a commandline to it. This will
     // make it obvious when a new tab/pane is created in a window.
-    _window->SummonWindow(false, 0);
+    _window->SummonWindow(summonArgs);
```
Comment (DHowett) [inline src/cascadia/WindowsTerminal/AppHost.cpp:598]: i know i asked you not to but also I don't care that much :smile:

File: src/cascadia/WindowsTerminal/IslandWindow.cpp [hunk 3]
```diff
@@ -1204,6 +1251,114 @@ void IslandWindow::_globalDismissWindow(const uint32_t dropdownDuration)
...
+// - activeMonitor: the monitor to move to.
+// Return Value:
+// - <none>
+void IslandWindow::_moveToMonitor(const MONITORINFO activeMonitor)
+{
+    // Get the monitor info for the window's current monitor.
+    const auto currentMonitor = _getMonitorForWindow(GetHandle());
+
+    til::rectangle currentRect{ currentMonitor.rcMonitor };
+    til::rectangle activeRect{ activeMonitor.rcMonitor };
```
Comment (DHowett) [inline src/cascadia/WindowsTerminal/IslandWindow.cpp:1345]: can these be const?

Comment (DHowett) [reply src/cascadia/WindowsTerminal/IslandWindow.cpp:1345]: can all of these be const actually?

**Review by DHowett — APPROVED** [review]
make sure you don't automerge the base! :smile:

**Pull request merged**